frontend/pulse3d/src/pages/upload-form.js

```diff
@@ -175,7 +175,11 @@ export default function UploadForm() {
     // checks if error value exists, no file is selected, or upload is in progress
     const checkConditions =
       !Object.values(paramErrors).every((val) => val.length === 0) ||
-      !((files.length > 0 && files[0] instanceof File) || (uploads && uploads.includes(files[0]))) ||
+      !(
+        (files.length > 0 && files[0] instanceof File) ||
+        (uploads && uploads.includes(files[0])) ||
+        defaultReanalysisFile
```

Comment:
We should try to unify the hooks for `files` and `defaultReanalysisFile` here. We're starting to go down the same road we did with IA where we're keeping two copies of the same state and it's causing problems. We can just extract `const [files, setFiles] = useState([]);` up to the parent component and set that directly instead of using this new `defaultReanalysisFile`. @luciipak does this seem like a better solution?

Reply:
Yeah I agree that there are already mechanisms set up to reset both the drag and drop zone and the input dropdown that would be easier to use than adding another state value. I'm not as up to speed on the bug, but the component should handle resetting itself. 
Additionally, I noticed that you're passing a `defaultFile` value down to the InputDropdownWidget, it's best to not add component specific props to a reusable component. I can tell that you're trying to set the default selected value to a specific file name... it would be better to add some initialValue  index prop instead so that other components can use it the same way.  

```
 useEffect(() => {
    // Set default file name if one is [present](url)
    if (defaultFile && defaultFile.length !== 0) {
      setSelected(defaultFile);
      handleSelection(options.indexOf(defaultFile));
    }
  }, [defaultFile]);
```

should be something like:
```
 useEffect(() => {
    if (initialValue) {
      setSelected(options[initialValue]);
      handleSelection(initialValue);
    }
  }, [initialValue]);
```

Reply:
I don't think the comments above were addressed in the PR changes.. removing use of defaultReanalysisFile and using existing mechanisms for resetting state

Reply:
I am using the given mechanisms to reset state, but there are none for resetting state when the page is navigated away from. The use effect in the `_app` is for resetting state of default file so that it is not present when the user attempts to do a normal reanalysis. I'm also not sure how I would remove `defaultReanalysisFile` without adding the selected file as a query param to the URL. 

The state for selected file is saved in the _app context and passed to children components who need it. When a user submits a reanalysis all built in mechanisms are used to reset state. But when you navigate away from the reanalysis page  without submitting the job and back, the default file still get set because it was never cleared in `_app.js`.  Is there a better way of keeping the state or passing the selected file for reanalysis ?

Reply:
My original point was that we know have `defaultReanalysisFile`, `setDefaultReanalysisFile`, and `const [files, setFiles] = useState([]);`. It's fine to have all these, but the default reanalysis file should only be used when initializing the component, i.e. something like this `const [files, setFiles] = useState(defaultReanalysisFile ? [defaultReanalysisFile] : []);`. That way it is used more appropriately as an initial value and not a duplicate copy of state that floats around in this component